containerize/Dockerfile

```diff
@@ -0,0 +1,52 @@
+ARG BUILDIMAGE
+FROM ${BUILDIMAGE} as builder
+
+ARG BUILD_DATE
+ARG ANALYSIS_NAME=analysis
+LABEL org.label-schema.build-date=$BUILD_DATE \
+      org.label-schema.name="$ANALYSIS_NAME Docker image" \
+      org.label-schema.description="Provides a runnable CMSSW image with $ANALYSIS_NAME pre-installed." \
+      org.label-schema.vendor="FNAL"
+
+USER cmsusr
+ARG USERDIR_BUILD=/home/${USER}
+WORKDIR ${USERDIR_BUILD}
+
+ARG CMSSW_VERSION
+ARG TAR
+
+COPY ${TAR} ${USERDIR_BUILD}/${CMSSW_VERSION}.tar.gz
+RUN pwd && \
+    ls -alh ./ && \
+    source /cvmfs/cms.cern.ch/cmsset_default.sh && \
+    echo "Unpacking ${CMSSW_VERSION}.tar.gz into ${PWD} ..." && \
+    tar -xzf ${CMSSW_VERSION}.tar.gz && \
+    rm ${CMSSW_VERSION}.tar.gz && \
+    ls -alh ./ && \
+    cd ${CMSSW_VERSION}/src/ && \
+    scram b ProjectRename && \
+    pwd && \
+    ls -alh ./ && \
+    echo "Setting the CMSSW environment ..." && \
+    eval `scramv1 runtime -sh` && \
+    echo "The CMSSW_BASE is ${CMSSW_BASE}" && \
+    cd ${HOME}
+
+# ---
+
+ARG BASEIMAGE
+FROM ${BASEIMAGE}
+
+ARG BUILD_DATE
+ARG ANALYSIS_NAME
+LABEL org.label-schema.build-date=$BUILD_DATE \
+      org.label-schema.name="$ANALYSIS_NAME Docker image" \
+      org.label-schema.description="Provides a runnable CMSSW image with $ANALYSIS_NAME pre-installed." \
+      org.label-schema.vendor="FNAL"
+
+ARG CMSSW_VERSION
+ARG NONPRIVILEGED_USER=cmsusr
+ARG USERDIR=/home/${NONPRIVILEGED_USER}
+WORKDIR ${USERDIR}
+COPY --from=builder --chown=cmsusr:cmsusr /home/cmsusr/${CMSSW_VERSION} ${USERDIR}/${CMSSW_VERSION}
```

Comment:
If `${USERDIR}` works here, can't you define `ARG USER_BUILD cmsusr` and then `USER ${USER_BUILD}`, etc. to work around the #19 issue?

Reply:
I can certainly do what you're suggesting, and it does partially get around the problem. However, it won't get around #19 because I still won't be able to evaluate arguments in the parameters (`chown`) of the `COPY` command. Let me know what you think now.

Reply:
I think this is an improvement. Can you update #19 to clarify the point about using arguments specifically in the parameters of the `COPY` command?

Reply:
Okay, #19 is updated.